datastore/datastore.go

```diff
@@ -0,0 +1,303 @@
+	Inventory       map[string]InventoryData `json:"inventory"`
+	Events          []EventData              `json:"events"`
+	Status          string                   `json:"status"`
+}
+
+//fields for stackdriver returns
+type StackdriverMetric struct {
+	TimeSeries []struct {
+		Metric struct {
+			Labels struct {
+				ApiMethod    string `json:"api_method"`
+				ResponseCode string `json:"response_code"`
+			} `json:"labels"`
+			Type string `json:"type"`
+		} `json:"metric"`
+		MetricKind string `json:"metricKind"`
+		Points     []struct {
+			Interval struct {
+				EndTime   time.Time `json:"endTime"`
+				StartTime time.Time `json:"startTime"`
+			} `json:"interval"`
+			Value struct {
+				Int64Value int64 `json:"int64Value,string"`
+			} `json:"value"`
+		} `json:"points"`
+		Resource struct {
+			Labels struct {
+				ModuleID  string `json:"module_id"`
+				ProjectID string `json:"project_id"`
+				VersionID string `json:"version_id"`
+			} `json:"labels"`
+			Type string `json:"type"`
+		} `json:"resource"`
+		ValueType string `json:"valueType"`
+	} `json:"timeSeries"`
+}
+
+type Client struct {
+	Dsc DatastoreImpl
+	Sdc StackDriverImpl
+}
+
+type DatastoreImpl interface {
+	GetAll(ctx context.Context, q *datastore.Query, dst interface{}) (keys []*datastore.Key, err error)
+}
+
+type StackDriverImpl interface {
+	List(name string) *monitoring.ProjectsTimeSeriesListCall
+}
+
+type KeyData struct {
+	AuthProviderX509CertURL string  `json:"auth_provider_x509_cert_url"`
+	AuthURI                 string  `json:"auth_uri"`
+	ClientEmail             string  `json:"client_email"`
+	ClientID                float64 `json:"client_id,string"`
+	ClientX509CertURL       string  `json:"client_x509_cert_url"`
+	PrivateKey              string  `json:"private_key"`
+	PrivateKeyID            string  `json:"private_key_id"`
+	ProjectID               string  `json:"project_id"`
+	TokenURI                string  `json:"token_uri"`
+	Type                    string  `json:"type"`
+}
+
+// InventoryData is the data type for inventory data produced by a plugin data
+// source and emitted to the agent's inventory data store
+type InventoryData map[string]interface{}
+
+// MetricData is the data type for events produced by a plugin data source and
+// emitted to the agent's metrics data store
+type MetricData map[string]interface{}
+
+// EventData is the data type for single shot events
+type EventData map[string]interface{}
+
+// NAME - name of plugin
+const NAME string = "datastore"
+
+// ProtocolVersion -
+const ProtocolVersion string = "1"
+
+func Run(log *logrus.Logger, prettyPrint bool, version string) {
+	var data = PluginData{
+		Name:            NAME,
+		ProtocolVersion: ProtocolVersion,
+		PluginVersion:   version,
+		Inventory:       make(map[string]InventoryData),
+		Metrics:         make([]MetricData, 0),
+		Events:          make([]EventData, 0),
+	}
+
+	var keyData KeyData
+
+	keyFile, err := ioutil.ReadFile("/var/secrets/google/key.json")
```

Comment:
It seems it would be better to stick with the standard method of connecting to datastore, https://github.com/GannettDigital/content-schema/blob/master/db/datastore.go#L113